wicker/core/storage.py

```diff
@@ -309,8 +313,11 @@ def _get_dataset_assets_path(self, dataset_id: DatasetID, prefix_to_trim: Option
             dataset_id.version,
             "assets",
         )
+        # prefer single return
         if prefix_to_trim:
-            return full_path.replace(prefix_to_trim, "")
+            full_path = full_path.replace(prefix_to_trim, "")
+        if self.mount_path:
```

Comment:
Thanks @pickles-bread-and-butter , I think this is a good start, but we probably need a cleaner user interface around this. It is a bit confusing and not very well defined 
1. how users should know `mount_path` would interact with `root_path` and why one should expect `prefix_to_trim` (`s3://`) should be decided based on `mount_path`
2. a lot of duplicate code on appending `full_path` or adjusting `prefix_to_trim` that we need to refactor in the same place

Reply:
1. Users don't interact with this function. It is hidden and only used internally. There is no public facing concern as it's explicitly stated as a private method. If a user is touching this function they are doing something wrong. As long as it commented enough for a dev that's sufficient.
2. It's a path cut, what you're suggesting would be mean we would be consolidating the s3 prefix cutting. IMO consolidating a one line function and a conditional is not worth it.

Reply:
@pickles-bread-and-butter I think maybe adding a short (global, public) helper function that accepts the input path, prefix_to_trim and mount path and does this operation might be good. This logic is repeated a bunch of times.

Reply:
Done, I made it static and private just fyi.